addons/mail/models/mail_link_preview_message.py

```diff
@@ -0,0 +1,37 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+from odoo.addons.mail.tools.discuss import Store
+
+
+class LinkPreviewMessage(models.Model):
+    _name = "mail.link.preview.message"
+    _inherit = "bus.listener.mixin"
+    _description = "Link between link previews and messages"
+
+    link_preview_id = fields.Many2one("mail.link.preview", index=True)
+    message_id = fields.Many2one("mail.message", index=True)
```

Comment:
Should have cascade, but it becomes harder to know when a link preview itself can be deleted. Especially as it can be used without message too.

Reply:
Link previews should not be deleted, they can always be reused for other messages or for `_search_or_create_from_url`.
It's a create once and reuse often scenario. Maybe we should even "refresh" the data at some point?

Reply:
Yes we will need a cache invalidation strategy in that case.